sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java

```diff
@@ -289,20 +291,133 @@ public RecognizeEntitiesResultCollection recognizeEntitiesBatch(
+     * For a list of enabled languages, check: <a href="https://aka.ms/talangs">this</a>
+     *
+     * <p><strong>Code Sample</strong></p>
+     * <p>Recognizes the PII entities details in a document with a provided language code.</p>
+     *
+     * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsClient.recognizePiiEntities#String-String}
+     *
+     * @param document The document to recognize PII entities details for.
+     * For text length limits, maximum batch size, and supported text encoding, see
+     * <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits">data limits</a>.
+     * @param language The 2 letter ISO 639-1 representation of language. If not set, uses "en" for English as default.
+     *
+     * @return The {@link PiiEntityCollection recognized PII entities collection}.
+     *
+     * @throws NullPointerException if {@code document} is null.
+     * @throws TextAnalyticsException if the response returned with an {@link TextAnalyticsError error}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public PiiEntityCollection recognizePiiEntities(String document, String language) {
+        Objects.requireNonNull(document, "'document' cannot be null.");
+        return client.recognizePiiEntities(document, language).block();
+    }
+
+    /**
+     * Returns a list of Personally Identifiable Information(PII) entities for the provided list of documents with
+     * provided language code and request options.
+     *
+     * <p><strong>Code Sample</strong></p>
+     * <p>Recognizes the PII entities details in a list of documents with a provided language code
+     * and request options.</p>
+     *
+     * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsClient.recognizePiiEntitiesBatch#Iterable-String-TextAnalyticsRequestOptions}
+     *
+     * @param documents A list of documents to recognize PII entities for.
+     * For text length limits, maximum batch size, and supported text encoding, see
+     * <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits">data limits</a>.
+     * @param language The 2 letter ISO 639-1 representation of language. If not set, uses "en" for English as default.
+     * @param options The {@link TextAnalyticsRequestOptions options} to configure the scoring model for documents
+     * and show statistics.
+     *
+     * @return A {@link RecognizePiiEntitiesResultCollection}.
+     *
+     * @throws NullPointerException if {@code documents} is null.
+     * @throws IllegalArgumentException if {@code documents} is empty.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public RecognizePiiEntitiesResultCollection recognizePiiEntitiesBatch(
+        Iterable<String> documents, String language, TextAnalyticsRequestOptions options) {
+        return client.recognizePiiEntitiesBatch(documents, language, options).block();
```

Comment:
This can call the method on L403.

Reply:
it is better to link to call async version of it rather than depends on other sync overload.